### PR TITLE
feat(artwork lists): improve name handling

### PR DESCRIPTION
**Pull request by anandaroop**
The type of this PR is: **Feat**

This PR solves [FX-4704] and [FX-4740]

### Description

This introduces two further client-side behaviors to the Artwork List name field:

- leading and trailing whitespaces are trimmed before submitting, on create (this behavior [already existed](https://github.com/artsy/force/blob/e4e9939edf84ce200774a9fa4cee4347eab20a66/src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditArtworkListModal.tsx#L57) on edit) — [89e2446](https://github.com/artsy/force/pull/12435/commits/89e2446ffc4fde78525d22935ff939c1b169bbfc)

- the name field is validated to be non-empty — [456e05b](https://github.com/artsy/force/pull/12435/commits/456e05bcbaced399a9ee054f095ae504c31f39f4)

The screencap below demonstrates an empty name being rejected, and then a name with  whitespace being trimmed:

https://github.com/artsy/force/assets/140521/5c2cbdab-0780-4123-b487-8aa517c323e4

[FX-4704]: https://artsyproduct.atlassian.net/browse/FX-4704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FX-4740]: https://artsyproduct.atlassian.net/browse/FX-4740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/CollectorProfile/Routes/Saves2/Components/ArtworkListForm/ArtworkListForm.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export const MAX_NAME_LENGTH = 40
 export const validationSchema = Yup.object().shape({
   name: Yup.string()
     .required(i18n.t("collectorSaves.artworkListForm.fields.name.required"))
+    .matches(/\S+/, { message: "name cannot be empty" })
```
Comment (anandaroop) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/ArtworkListForm/ArtworkListForm.tsx:17]: Keeping this new validation simple: at least one non-whitespace character is required, anywhere in the name.

Comment (rajsam003) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/ArtworkListForm/ArtworkListForm.tsx:17]:
```suggestion
    .matches(/\S+/, { message: "Name cannot be empty" })
```

File: src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/__tests__/EditArtworkListModal.jest.tsx [hunk 2]
```diff
@@ -126,6 +126,26 @@ describe("EditArtworkListModal", () => {
...
+    const { nameInputField, saveButton } = setup()
+
+    fireEvent.change(nameInputField, {
+      target: { value: "   " },
+    })
+
+    fireEvent.click(saveButton)
+
+    await waitFor(() => {
+      expect(screen.getByText("name cannot be empty")).toBeInTheDocument()
```
Comment (rajsam003) [inline src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/__tests__/EditArtworkListModal.jest.tsx:145]:
```suggestion
      expect(screen.getByText("Name cannot be empty")).toBeInTheDocument()
```

**Review by rajsam003 — DISMISSED** [review]
👏

File: src/Apps/CollectorProfile/Routes/Saves2/Components/ArtworkListForm/ArtworkListForm.tsx (hunk 1 above)
Comment (anandaroop) [reply src/Apps/CollectorProfile/Routes/Saves2/Components/ArtworkListForm/ArtworkListForm.tsx]: Good call, and it reminded me to move this hard-coded text into the translations file instead: https://github.com/artsy/force/pull/12435/commits/1b01618ceea2820c8b5a8a85692de7059f3c5468

**Pull request merged**